ui/app/src/components/ClusterManagement/ClusterManagement.tsx

```diff
@@ -107,19 +65,17 @@ export function ClusterManagement() {
           </Tooltip>
         }
       />
-      <SuspenseWithEmptyState
-        resource={resource}
-        suspenseProps={{
-          fallback: <ClustersGridSkeletonTable />
-        }}
-        withEmptyStateProps={{
-          emptyStateProps: {
-            title: <FormattedMessage id="clusterGrid.emptyStateMessage" defaultMessage="No Clusters Found" />
-          }
-        }}
-      >
-        <ClusterGridUI resource={resource} onDeleteCluster={onDeleteCluster} />
-      </SuspenseWithEmptyState>
+      {error ? (
+        <ApiResponseErrorState error={error} />
+      ) : fetching ? (
+        <LoadingState />
+      ) : clusters?.length ? (
+        <ClusterGridUI clusters={clusters} />
+      ) : (
+        <EmptyState
+          title={<FormattedMessage id="clusterGrid.emptyStateMessage" defaultMessage="No Clusters Found" />}
+        />
+      )}
```

Comment:
Avoid showing empty state when the component bootstraps. Test appropriately all paths.

```suggestion
      ) : clusters ? (
        clusters.length ? <ClusterGridUI clusters={clusters} /> :         <EmptyState
          title={<FormattedMessage id="clusterGrid.emptyStateMessage" defaultMessage="No Clusters Found" />}
        />

      )}
```

Reply:
should it be like
```
error 
   ? ErrorState
   : clusters 
      ? clusters.length 
          ? ClusterGrid
          : EmptyState
      : LoadingState
  ```    
?